setools/policyrep/terule.pxi

```diff
@@ -464,17 +464,18 @@ cdef class FileNameTERule(BaseTERule):
         readonly str filename
 
     @staticmethod
-    cdef inline FileNameTERule factory(SELinuxPolicy policy, sepol.filename_trans_t *key,
-                                       sepol.filename_trans_datum_t *datum):
+    cdef inline FileNameTERule factory(SELinuxPolicy policy,
+                                       sepol.filename_trans_key_t *key,
+                                       Type stype, size_t otype):
```

Comment:
`stype` can only be a type and never an attribute?

Reply:
Yes, filename transitions are always expanded in the binary policy. I tried to add support for attributes there once, but I'd have to prove that it doesn't break performance in the kernel (looking up rules for types that belong to many attributes can be quite costly), which seemed like too much hassle at that time, so I gave up on that eventually.